src/InputHandling/TrickInputHandler.cpp

```diff
@@ -1,28 +1,18 @@
 #include "InputHandling/TrickInputHandler.hpp"
 
-DEFINE_TYPE(TrickSaber::InputHandling, TrickInputHandler);
+
 
 namespace TrickSaber::InputHandling {
-    void TrickInputHandler::ctor() {
-        trickHandlerSets = TrickHandlerSetsDictionary::New_ctor();
+    TrickInputHandler::TrickInputHandler() = default;
 
-        trickHandlerSets->Add(TrickSaber::TrickAction::Throw, TrickHandlerHashSet::New_ctor());
-        trickHandlerSets->Add(TrickSaber::TrickAction::Spin, TrickHandlerHashSet::New_ctor());
-    }
-    
-    void TrickInputHandler::Add(TrickSaber::TrickAction action, InputHandler* handler) {
+    void TrickInputHandler::Add(TrickSaber::TrickAction action, std::unique_ptr<InputHandler> handler) {
         if (action == TrickSaber::TrickAction::None) return;
-        TrickHandlerHashSet* set = nullptr;
-        if (trickHandlerSets->TryGetValue(action, byref(set))) {
-            set->Add(handler);
-        }
+
+        trickHandlerSets[action].emplace_back(std::move(handler));
```

Comment:
Pretty sure doing something like this is invalid because the given action doesn't yet exist in that map

Reply:
it is not invalid, the map emplaces it when using `[]` if it has a default constructor